contracts/Optimism_SpokePool.sol

```diff
@@ -87,14 +97,24 @@ contract Optimism_SpokePool is CrossDomainEnabled, SpokePoolInterface, SpokePool
     }
 
     function _bridgeTokensToHubPool(DestinationDistributionLeaf memory distributionLeaf) internal override {
-        // TODO: Handle WETH token unwrapping
+        address l2Token = distributionLeaf.l2TokenAddress;
+        address l1Destination = hubPool;
+        // If the token being bridged is WETH then we need to first unwrap it to ETH and then send ETH over the
+        // canonical bridge. On Optimism, this is address 0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000.
+        if (distributionLeaf.l2TokenAddress == address(weth)) {
+            WETH9(l2Token).withdraw(distributionLeaf.amountToReturn); // Unwrap ETH.
+            l2Token = l2Eth; // Set the bridged token to the Optimism ETH address.
+            l1Destination = l1EthWrapper; // Set the destination to the L1 ETH wrapper.
```

Comment:
Why is the destination the l1 eth wrapper, how does thhis wrapper contract know who the end recipient is?

Reply:
this will be this contract: https://github.com/across-protocol/across-smart-contracts/blob/master/contracts/insured-bridge/ovm/Optimism_Wrapper.sol which simply receives eth, wraps it and sends it to the hupPool (the destination). I'll add this contract to this PR.

Reply:
I've added a modified version of this contract to this PR.

Reply:
@nicholaspai  I've pretty drastically refactored this.

Reply:
ah i see